src/components/TradeWidget/Price.tsx

```diff
@@ -225,10 +227,10 @@ const Price: React.FC<Props> = ({
   return (
     <Wrapper>
       <strong>
-        Limit Price <OrderBookBtn baseToken={receiveToken} quoteToken={sellToken} />
+        Limit Price <OrderBookBtn baseToken={baseToken} quoteToken={quoteToken} />
       </strong>
       {/* using display: none to hide to avoid hook-form reregister */}
-      <PriceInputBox hidden={priceShown !== 'DIRECT'}>
+      <PriceInputBox hidden={(isPriceInverted && wasPriorityAdjusted) || (!isPriceInverted && !wasPriorityAdjusted)}>
```

Comment:
I think the logic is a bit complex, instead of complicating when rendering the view, can we just use "aux" variables for the quote/base returned by the function, and make the final selection of quote/base based on that plus `isPriceInverted`. 

This way, you move the logic to the JS part